website/addons/base/serializer.py

```diff
@@ -156,6 +144,16 @@ def user_is_owner(self):
     def credentials_owner(self):
         return self.node_settings.user_settings.owner
 
+    @property
+    def serialized_node_settings(self):
+        result = super(OAuthAddonSerializer, self).serialized_node_settings
+        if self.node_settings.oauth_provider.short_name != 'dataverse':
```

Comment:
As discussed privately, there isn't a quick correct way to fix this. Consider architectural review.